src/msg/async/rdma/Infiniband.cc

```diff
@@ -470,8 +470,7 @@ Infiniband::ProtectionDomain::ProtectionDomain(CephContext *cct, Device *device)
 
 Infiniband::ProtectionDomain::~ProtectionDomain()
 {
-  int rc = ibv_dealloc_pd(pd);
-  assert(rc == 0);
+  ibv_dealloc_pd(pd);
```

Comment:
Only commits starting from the https://github.com/Mellanox/ceph/pull/1/commits/793d73937fe51b521b9012ee61ebd5e18f45f0b2 are relevant

assert() is removed because it is possible to get a bad exit code (resources busy) if client app is terminated by the signal (Ctrl-C). It is easier to let OS do a cleanup than trying to do a proper cleanup in a signal handler